packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_masternode_update.test.ts

```diff
@@ -757,4 +758,208 @@ describe('UpdateMasternode', () => {
+        ownerAuthAddress: collateralAddress
+      }))
+    }
+
+    await container.generate(45)
+
+    {
+      const masternodes = await jsonRpc.masternode.listMasternodes()
+      expect(masternodes[masternodeId]).toStrictEqual(expect.objectContaining({
+        state: 'ENABLED',
+        ownerAuthAddress: collateralAddress
+      }))
+    }
+  })
+})
+
+describe('Update Masternode (Multi-containers)', () => {
+  let tGroup: TestingGroup
+  const node: Testing[] = []
+  let builderNode1: P2WPKHTransactionBuilder
+  let providersNode1: MockProviders
+
+  beforeAll(async () => {
+    tGroup = TestingGroup.create(2, i => new MasterNodeRegTestContainer(RegTestFoundationKeys[i]))
+    await tGroup.start()
+    node.push(tGroup.get(0))
+    node.push(tGroup.get(1))
+    await node[0].container.waitForWalletCoinbaseMaturity()
+    await node[0].container.waitForWalletBalanceGTE(1000)
+    providersNode1 = await getProviders(node[1].container)
+
+    await providersNode1.randomizeEllipticPair()
+    builderNode1 = new P2WPKHTransactionBuilder(providersNode1.fee, providersNode1.prevout, providersNode1.elliptic, RegTest)
+    await fundEllipticPair(node[1].container, providersNode1.ellipticPair, 10)
+    await providersNode1.setupMocks()
+
+    await node[0].container.generate(1)
+    await node[1].container.generate(1)
+  })
+
+  afterAll(async () => {
+    await tGroup.stop()
+  })
+
+  it('should throw error if incorrect authorization is provided', async () => {
+    await node[0].container.waitForWalletCoinbaseMaturity()
+    await node[1].container.waitForWalletCoinbaseMaturity()
+
+    // enable updating
+    await node[0].rpc.masternode.setGov({
+      ATTRIBUTES: {
+        'v0/params/feature/mn-setowneraddress': 'true',
+        'v0/params/feature/mn-setoperatoraddress': 'true',
+        'v0/params/feature/mn-setrewardaddress': 'true'
+      }
+    })
+    await node[0].generate(1)
+
+    const masternodeOwnerAddress = await node[0].rpc.wallet.getNewAddress()
+    const masternodeId = await node[0].rpc.masternode.createMasternode(masternodeOwnerAddress)
+    await node[0].generate(100) // create masternode and wait for it to be enabled
+
+    const operatorAddress = await node[0].rpc.wallet.getNewAddress()
+    const operatorAddressDest: P2WPKH = P2WPKH.fromAddress(RegTest, operatorAddress, P2WPKH)
+    const operatorAddressDestKeyHash = operatorAddressDest.pubKeyHash
+    const operatorScript = await fromAddress(operatorAddress, 'regtest')?.script as Script
+    await node[0].generate(4)
+    await tGroup.waitForSync() // container2 should know about the new masternode
+
+    const updateMasternode: UpdateMasternode = {
+      nodeId: masternodeId,
+      updates: [
+        {
+          updateType: 0x02,
+          address: { addressType: 0x04, addressPubKeyHash: operatorAddressDestKeyHash }
+        }
+      ]
+    }
+    const txn: TransactionSegWit = await builderNode1.masternode.update(updateMasternode, operatorScript)
```

Comment:
nit: Might improve readability if the 0s (and 1s) are labeled throughout.

```suggestion
    const masternodeOwnerAddress0 = await node[0].rpc.wallet.getNewAddress()
    const masternodeId0 = await node[0].rpc.masternode.createMasternode(masternodeOwnerAddress0)
    await node[0].generate(100) // create masternode and wait for it to be enabled

    const operatorAddress0 = await node[0].rpc.wallet.getNewAddress()
    const operatorAddressDest0: P2WPKH = P2WPKH.fromAddress(RegTest, operatorAddress, P2WPKH)
    const operatorAddressDestKeyHash0 = operatorAddressDest.pubKeyHash
    const operatorScript0 = await fromAddress(operatorAddress, 'regtest')?.script as Script
    await node[0].generate(4)
    await tGroup.waitForSync() // container2 should know about the new masternode

    const updateMasternode0: UpdateMasternode = {
      nodeId: masternodeId0,
      updates: [
        {
          updateType: 0x02,
          address: { addressType: 0x04, addressPubKeyHash: operatorAddressDestKeyHash0 }
        }
      ]
    }
    const txn: TransactionSegWit = await builderNode1.masternode.update(updateMasternode0, operatorScript0)
```

Reply:
Amended